### PR TITLE
Add create_dir 

### LINKED ISSUE #48: Add missing filesystem operations
It would be great if we could add the most important functions from `tokio::fs`/`std::fs` to this crate. As far as I can tell `tokio-uring` only exposes `fs::File` right now and none of the ones below are supported yet.

Directories:
- [x] create_dir
- [ ] read_dir
- [x] remove_dir
- [ ] remove_dir_all

Files:
- [ ] metadata
- [x] rename
- [ ] copy
- [x] remove_file

Links (I don't really need these but would be great to have):
- [ ] symlink_metadata
- [ ] soft_link
- [ ] hard_link
- [ ] read_link

This should be fairly straightforward to implement, right? I can try to take a stab at some of these once I find some time.

### PR DESCRIPTION
**Pull request by kaimast**
This adds a `create_dir` function call to `tokio_uring::fs` and partially fixes #48.

I also renamed `tests/directory.rs` to `tests/fs_directory.rs` so the name is in sync with that of `tests/fs_file.rs`.

### CODE REVIEW

File: src/io/mkdir_at.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        })
+    }
+}
+
+impl Completable for Mkdir {
+    type Output = io::Result<()>;
+
+    fn complete(self, cqe: CqeResult) -> Self::Output {
+        cqe.result?;
+        Ok(())
```
Comment (FrankReh) [inline src/io/mkdir_at.rs:38]: Also, like unlink_at.rs,
```rust
cqe.result.map(|_| ())
```

**Review by FrankReh — CHANGES_REQUESTED** [review]
Nice. Just minor stuff.

**Comment by FrankReh** [comment]
@kaimast Would it be alright with you if I made the two tweaks and then merged?

**Comment by kaimast** [comment]
Yes, of course.

**Comment by kaimast** [comment]
I made the changes and also resolved conflicts with more recent changes in master.

**Review by FrankReh — APPROVED** [review]
Looks good. Thanks for getting to this.

**Pull request merged**